internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -1369,7 +1369,13 @@ func resourceKubernetesClusterUpdate(d *pluginsdk.ResourceData, meta interface{}
 
 	if d.HasChange("run_command_enabled") {
 		updateCluster = true
-		existing.ManagedClusterProperties.APIServerAccessProfile.DisableRunCommand = utils.Bool(!d.Get("run_command_enabled").(bool))
+		if existing.ManagedClusterProperties.APIServerAccessProfile != nil {
+			existing.ManagedClusterProperties.APIServerAccessProfile.DisableRunCommand = utils.Bool(!d.Get("run_command_enabled").(bool))
+		} else {
+			existing.ManagedClusterProperties.APIServerAccessProfile = &containerservice.ManagedClusterAPIServerAccessProfile{
+				DisableRunCommand: utils.Bool(!d.Get("run_command_enabled").(bool)),
+			}
+		}
```

Comment:
fwiw most of the other resources tend to do this - which means we only have a single assignment to this property:

```suggestion
		if existing.ManagedClusterProperties.APIServerAccessProfile == nil {
			existing.ManagedClusterProperties.APIServerAccessProfile = &containerservice.ManagedClusterAPIServerAccessProfile{
			}
		}
		existing.ManagedClusterProperties.APIServerAccessProfile.DisableRunCommand = utils.Bool(!d.Get("run_command_enabled").(bool))
```